Dev guide: mention field types for manifest, along with examples edited to the new format

josephsl opened the pull request:
### Link to issue number:
Fixes #8593

### Summary of the issue:
Clarify proper value types for add-on manifest fields.

### Description of how this pull request fixes the issue:
In add-on manifest section in developer guide, add value types for manifest fields. Also, added a note regarding enclosing summary and description data in quotes, and for multi-line descriptions, advised readers to use triple quotes.

### Testing performed:
Compiled developer guide.

### Known issues with pull request:
None

### Change log entry:
None

josephsl committed:
Dev guide: mention field types for manifest, along with examples edited to show proper value format. Re #8593

jscholes commented:
In the example manifest, the author field isn't enclosed in quotes as you state it should be.

josephsl commented:
Hi, right, I’ll address that in the next commit. Thanks.

 

From: James Scholes <notifications@github.com> 
Sent: Sunday, August 5, 2018 10:44 AM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Author <author@noreply.github.com>
Subject: Re: [nvaccess/nvda] Dev guide: mention field types for manifest, along with examples edited to the new format (#8594)

 

In the example manifest, the author field isn't enclosed in quotes as you state it should be.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/8594#issuecomment-410536061> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AHgLkOqEEZdhnEhDV2eBENNK72fn4fYJks5uNy71gaJpZM4Vvdhq> .

josephsl committed:
Dev guide: put manifest/author in quotes. Re #8593.

Reviewed by James Scholes: contrary to what the spec says, add-on author value in manifest is not quoted, so quote this.

derekriemer reviewed (dismissed):
Make sure to quote all string fields. Also, see other comments.

jscholes commented:
Agree with @derekriemer on this one.  For the most part the programming concept of a string calls for it to be quoted.  Usually, there is no such thing as a string which isn't quoted and the only thing allowing it in this case is the fact that the INI file format and parser don't care.  It would be more straight forward to just tell people to quote all strings in their manifest, including the internal add-on name.  If NVDA itself can't handle that, the abstraction is leaking.

At the very least, it would be clearer to remove the "enclosed in quotes" bit.  Have the type "string" in brackets as you have here, but if quoting is truly only appropriate for certain fields, write something like:

> This value must be enclosed in quotes."

At the end of the field description.

Example:

> summary (string): The name of the add-on as shown to the user. This value must be enclosed in quotes. (Mandatory)

Re: email addresses: presumably Mick originally wrote this and is happy for his address to be there, but it doesn't really matter either way.

josephsl committed:
Dev guide manifest section: address review actions. Re #8593.

Reviewed by Derek Riemer and James Scholes: when mentioning string type, also mention the fact that it must be enclosed in quotes. For the example manifest, quoted all striong values. In addition, docFileName now has an example.

michaelDCurran commented:
If you are aiming for 2018.3 then the base branch must be now beta.

josephsl commented:
Hi, shall be done. P.S. Can you let developers know when that happens so PR writers can rebase their work on beta please? Thanks.

 

From: Michael Curran <notifications@github.com> 
Sent: Sunday, August 5, 2018 5:05 PM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Author <author@noreply.github.com>
Subject: Re: [nvaccess/nvda] Dev guide: mention field types for manifest, along with examples edited to the new format (#8594)

 

If you are aiming for 2018.3 then the base branch must be now beta.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/8594#issuecomment-410558191> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AHgLkBAiwzxWpMU8vZBFB3nN5M2cW6DEks5uN4gkgaJpZM4Vvdhq> .

michaelDCurran commented:
This is on a case by case basis.

The developer guide is fine. All other PRs, unless they are fixing a 
regression after 2018.2.1, should be still based on master and will not 
be in 2018.3.

josephsl commented:
Hi,

Note that for now, beta and master are the same, so when I do git rebase beta on my branch, it shows that it is up to date.

Thanks.

michaelDCurran commented:
That is right. But you need to change the base branchin the pr itself on 
Github.

josephsl commented:
Hi,

For future reference: to change the base branch, select edit title, select change base, then select the new base branch. When asked to confirm, say "change".

Thanks.

josephsl commented:
Hi, done. Thanks.

 

From: Michael Curran <notifications@github.com> 
Sent: Sunday, August 5, 2018 5:10 PM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Author <author@noreply.github.com>
Subject: Re: [nvaccess/nvda] Dev guide: mention field types for manifest, along with examples edited to the new format (#8594)

 

That is right. But you need to change the base branchin the pr itself on 
Github.




—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/8594#issuecomment-410558535> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AHgLkOIiP0mFbd7r497zRjH7cyHOi_VQks5uN4lxgaJpZM4Vvdhq> .

feerrenrut commented on developerGuide.t2t:
```diff
@@ -654,24 +654,25 @@ This manifest file contains key = value pares declaring info such as the add-on'
-- description: A sentence or two describing what the add-on does.
-- url: A URL where this add-on, further info and upgrades can be found.
-- docFileName: The name of the main documentation file for this add-on; e.g. readme.html. See the [Add-on Documentation #AddonDoc] section for more details.
+- name (string enclosed in quotes): A short unique name for the add-on. This is used to differentiate add-ons internally and is not shown to the user. (Mandatory)
```
I would prefer if this just said string. Then a note underneath: "strings should be enclosed in quotes, (see examples)"

josephsl committed:
Merged master

josephsl committed:
Dev guide: address review comments.

Reviewed by Reef Turner (NV Access): add a note that strings must be enclosed in quotes. Edited example manifest to enclose all string values in quotes, including minimum and last tested NVDA versions.

feerrenrut reviewed (approved):
Thanks for addressing my concern about the repetition. The PR was targeting beta, I think it would be ok to merge this into into RC to be included in the 2019.1.1 release, but it will need to be rebased on RC very quickly.

feerrenrut commented:
Actually, if this is delivered to RC it won't be translated. I will merge this to master now.

lukaszgo1 commented:
Sorry for going slightly offtopic here, but is Dev guide translated at all?

Adriani90 commented:
Yes it is, but I am not sure which version. I don't remember any revision being submited to the translation system during last 2 or 3 years. The version on the translation system might be outdated.

josephsl commented:
Hi, the “dev guide” that folks will read in other languages is not this one – it is the add-on dev guide that has been translated. Thanks.

 

From: Adriani90 <notifications@github.com> 
Sent: Monday, April 8, 2019 6:02 AM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Author <author@noreply.github.com>
Subject: Re: [nvaccess/nvda] Dev guide: mention field types for manifest, along with examples edited to the new format (#8594)

 

Yes it is, but I am not sure which version. I don't remember any revision being submited to the translation system during last 2 or 3 years. The version on the translation system might be outdated.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/8594#issuecomment-480822242> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AHgLkCAYttUE8P46QK_MtDaeqPK5jHVnks5vez3igaJpZM4Vvdhq> .
